frontend/src/core/entities/reducer.js

```diff
@@ -51,10 +55,36 @@ function updateEntityTranslation(
     });
 }
 
+function injectSiblingEntities(
+    entities: Entities,
+    siblings: EntitySiblings,
+    entity: number,
+) {
+    const copy = [...entities];
+
+    const index = entities.findIndex((item) => item.pk === entity);
```

Comment:
Now that you have index, how about you try this approach:

1. Create a [concatenated list](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat) of preceding siblings, entity item (i.e. `entities[index]`) and succeeding siblings.
2. Replace entity item in the entities array with a concatenated list - `entities.splice(index, 1, concatenated);`.
3. Return a [flat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) list with `entities.flat()`.

I also don't think you'll need to create a `copy`.

Reply:
You are a wizard